File: src/engine/player.py
from lib import contract

from . import unit, consts


class Player(object):
    @contract.self_accepts(int, consts.Nation)
    def __init__(self, player_id, nation_):
        self.player_id = player_id
        self.nation = nation_
        self.team = consts.Team(nation_)

    @contract.self_accepts(bool)
    def serialize(self, public):
        return {
            "player_id": self.player_id,
            "nation": self.nation.serialize(public),
            "team": self.team.serialize(public)
        }

File: src/engine/game/player.go
package game

import (
	"api"
)

type Player struct {
	playerId int
	nation   nation
	team     team
}

func NewPlayer(playerId int, nation nation, team team) *Player {
	return &Player{playerId: playerId, nation: nation, team: team}
}

func (player *Player) Serialize() *api.PlayerStruct {
	return &api.PlayerStruct{
		Nation: int(player.nation),
		Team:   int(player.team)}
}

func (player *Player) GetPlayerId() int {
	return player.playerId
}

func (player *Player) GetNation() int {
	return int(player.nation)
}
